app/services/rules_cleanup_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# This class cleans up unused rules and reimports
+class RulesCleanupService
+  class << self
+    def run!
+      # Clean up rules that don't belong to any profile
+      Rule.where.not(id: ProfileRule.select(:rule_id)).delete_all
+
+      # Clean up rules that belong only to canonical profiles
+      delete_rules(select_rules)
+
+      # Reimport all rules if we're not in the testing environment
+      Rake::Task['ssg:import_rhel_supported'].execute unless Rails.env.test?
+    end
+
+    def select_rules
+      profiles = Profile.canonical(false).select(:parent_profile_id).distinct
```

Comment:
Oh, this is the line that selects canonical profiles that have a child. In other words "used profiles". This is also the reason that requires reimport.